api/src/main/java/org/ccci/idm/user/User.java

```diff
@@ -411,12 +412,15 @@ public void setCruProxyAddresses(Collection<String> cruProxyAddresses) {
         this.cruProxyAddresses = cruProxyAddresses;
     }
 
-    public Collection<String> getCruPasswordHistory() {
-        return cruPasswordHistory;
+    public Set<String> getCruPasswordHistory() {
+        return ImmutableSet.copyOf(cruPasswordHistory);
     }
 
-    public void setCruPasswordHistory(final Collection<String> cruPasswordHistory) {
-        this.cruPasswordHistory = cruPasswordHistory;
+    public void setCruPasswordHistory(@Nullable final Collection<String> history) {
+        cruPasswordHistory.clear();
+        if (history != null) {
+            cruPasswordHistory.addAll(history);
```

Comment:
for the admin webflow we store a copy of the `User` object in the flow scope so we can use it for rendering without needing to reload the user on every request. This data is persisted across requests using object serialization.